---
### Inference Config
ROS_INPUT: /camera/color/image_raw # ROS Input Image topic
CV_INPUT: 0                    # OpenCV Readable Input for py-scripts
VISUALIZE: True                # Disable for performance increase
CPU_ONLY: False                # CPU Placement for speed test
USE_OPTIMIZED: False           # whether to use the optimized model (only possible if transform with script)


### Testing
IMAGE_PATH: 'test_images'      # path for test.py test_images
LIMIT_IMAGES: None             # if set to None, all images are used
WRITE_TIMELINE: True           # write json timeline file (slows infrence)
SEQ_MODELS: []                 # List of Models to sequentially test (Default all Models)


### Object_Detection
WIDTH: 600                     # OpenCV only supports 4:3 formats others will be converted
HEIGHT: 600                    # 600x600 leads to 640x480
MAX_FRAMES: 5000               # only used if visualize==False
FPS_INTERVAL: 5                # Interval [s] to print fps of the last interval in console
PRINT_INTERVAL: 500            # intervall [frames] to print detections to console
PRINT_TH: 0.5                  # detection threshold for det_intervall
## speed hack
SPLIT_MODEL: False              # Splits Model into a GPU and CPU session (currently only works for ssd_mobilenets)
SSD_SHAPE: 300                 # used for the split model algorithm (currently only supports ssd networks trained on 300x300 and 600x600 input)
## Tracking
USE_TRACKER: False             # Use a Tracker (currently only works properly WITHOUT split_model)
TRACKER_FRAMES: 20             # Number of tracked frames between detections
NUM_TRACKERS: 5                # Max number of objects to track
## Model
OD_MODEL_NAME: 'mask_rcnn_mobilenet_v1_224_coco_person_2M_4'
OD_MODEL_PATH: 'mask_rcnn_mobilenet_v1_224_coco_person_2M_4/{}'
LABEL_PATH: 'tf_coco_label_map.pbtxt'
NUM_CLASSES: 1


### DeepLab
ALPHA: 0.3                     # mask overlay factor
BBOX: True                     # compute boundingbox in postprocessing
MINAREA: 500                   # min Pixel Area to apply bounding boxes (avoid noise)
## Model
DL_MODEL_NAME: 'deeplabv3_mnv2_pascal_train_aug_2018_01_29'
DL_MODEL_PATH: 'deeplabv3_mnv2_pascal_train_aug/{}'
